File: FootballApp/Models/Match.cs
using System;
using System.Collections.Generic;

namespace FootballApp.Models
{
    public class Match
    {
        public int Id { get; set; }
        public int HomeTeamId { get; set; }
        public Team HomeTeam { get; set; }
        public int AwayTeamId { get; set; }
        public Team AwayTeam { get; set; }
        public DateTime DateOfMatch { get; set; }
        public DateTime TimeOfMatch { get; set; }
        public int HomeGoals { get; set; }
        public int AwayGoals { get; set; }
        public int FixtureId { get; set; }
        public Fixture Fixture { get; set; }
        public bool HasEnded { get; set; }
        public ICollection<Goal> Goals { get; set; }
        public ICollection<Card> Cards { get; set; }
    }
}

File: FootballApp-SPA/src/app/_models/Match.ts
import { Team } from './team';
import { Fixture } from './Fixture';

export interface Match {
    id: number;
    homeTeamId: number;
    homeTeam: Team;
    awayTeamId: number;
    awayTeam: Team;
    dateOfMatch: Date;
    timeOfMatch: Date;
    homeGoals: number;
    awayGoals: number;
    fixtureId: number;
    fixture: Fixture;
    hasEnded: boolean;
}
